keps/sig-node/2681-pod-host-ip/README.md

```diff
@@ -0,0 +1,559 @@
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The proposal aims to improve the Pod's ability to obtain the address of the node
+
+## Motivation
+
+### Goals
+
+- Field `status.hostIPs` added for Pod
+- Downward API support for `status.hostIPs`
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+In some scenarios, the application needs to know the both IPv4 and IPv6 addresses.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Versioned API Change: PodStatus v1 core
+
+In order to maintain backwards compatibility for the core V1 API, this proposal
+retains the existing (singular) "HostIP" field in the core V1 version of the
+PodStatus V1 core API and adds a new array of structures that store host IPs
+along with associated metadata for that IP.
+
+``` go
+    // HostIP address of the host to which the pod is assigned. Empty if not yet scheduled.
+    // +optional
+    HostIP string
+
+    // HostIP represents the IP address of a host.
+    // IP address information. Each entry includes:
+    //    IP: An IP address allocated to the host.
+    type HostIP struct {
+      // ip is an IP address (IPv4 or IPv6) assigned to the host
+      IP string
+    }
+
+    // HostIPs holds the IP addresses allocated to the host. If this field is specified, the 0th entry must
+    // match the hostIP field. This list is empty if no IPs have been allocated yet.
+    // +optional
+    HostIPs []HostIP
+```
+
+### PodStatus Internal Representation
+
+PodStatus internally indicates that the original use of HostIP will remain unchanged,
+and HostIPs is added for subsequent use
+
+``` go
+    // HostIP address information for entries in the (plural) HostIPs field.
+    // Each entry includes:
+    //    IP: An IP address allocated to the host.
+    type HostIP struct {
+      // ip is an IP address (IPv4 or IPv6) assigned to the host
+      IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
+    }
+
+    // HostIPs holds the IP addresses allocated to the host. If this field is specified, the 0th entry must
+    // match the hostIP field. This list is empty if no IPs have been allocated yet.
+    // +optional
+    // +patchStrategy=merge
+    // +patchMergeKey=ip
+    HostIPs []HostIP `json:"hostIPs,omitempty" protobuf:"bytes,14,rep,name=hostIPs" patchStrategy:"merge" patchMergeKey:"ip"`
+```
+
+### Maintaining Compatible Interworking between Old and New Clients
```

Comment:
For the record:

There's the "set values on read" part so the new field will always be populated even on old, stored objects. 
 See pkg/registry/core/service/storage/storage.go `defaultOnRead()`

Then there's the "figure out what the user meant and sync the fields on write" part.  See pkg/registry/core/service/strategy.go `PrepareForCreate()` and `PrepareForUpdate()` -> `NormalizeClusterIPs()`